src/Sentry/Internal/MainExceptionProcessor.cs

```diff
@@ -103,7 +103,7 @@ internal IEnumerable<SentryException> CreateSentryException(Exception exception)
                 Type = exception.GetType()?.FullName,
                 Module = exception.GetType()?.Assembly?.FullName,
                 Value = exception.Message,
-                ThreadId = Thread.CurrentThread.ManagedThreadId,
+                ThreadId = Environment.CurrentManagedThreadId,
```

Comment:
Shouldn't we add a test that invokes MainExceptionProcessor and MainSentryEventProcessor to check if ThreadId and SentryThread.Id are equal?

Other than that, the PR looks good for me.